File: src/191. Number of 1 Bits/191.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n:
            n = n & (n-1)
            count += 1
        return count


if __name__ == "__main__":
    n = 127
    print(Solution().hammingWeight(n))

File: src/191. Number of 1 Bits/191.go
func hammingWeight(num uint32) int {
    count := 0
    for num > 0 {
        num = num & (num-1)
        count ++
    }
    return count  
}

func hammingWeight(num uint32) int {
    var count int
    for num !=0{
        if num & 1 == 1{
            count++
        }
        num >>= 1
    }
    return count
}
